src/wallet/wallettool.cpp

```diff
@@ -28,9 +28,11 @@ static void WalletCreate(CWallet* wallet_instance)
 
     // generate a new HD seed
     wallet_instance->SetupLegacyScriptPubKeyMan();
-    // NOTE: we do not yet create HD wallets by default
-    // auto spk_man = wallet_instance->GetLegacyScriptPubKeyMan();
-    // spk_man->GenerateNewHDChain("", "");
+    auto spk_man = wallet_instance->GetLegacyScriptPubKeyMan();
+    // NOTE: drop this condition after removing option to create non-HD wallets
+    if (spk_man->IsHDEnabled()) {
+        spk_man->GenerateNewHDChain("", "");
```

Comment:
_Note: This review comment is not a blocker for this pull request_

Minor nit

```c++
        spk_man->GenerateNewHDChain("" /* secureMnemonic */, "" /* secureMnemonicPassphrase */);
```

Reply:
Need to use correct format at least :)
```
        spk_man->GenerateNewHDChain(/*secureMnemonic=*/"", /*secureMnemonicPassphrase=*/"" );
```

Reply:
To be fair, we seem to use [every](https://github.com/dashpay/dash/blob/ce67e07294c90bcd9a0ffb038bf8aaad0670775e/src/wallet/rpcdump.cpp#L255) [possible](https://github.com/dashpay/dash/blob/ce67e07294c90bcd9a0ffb038bf8aaad0670775e/src/test/addrman_tests.cpp#L421) [variant](https://github.com/dashpay/dash/blob/ce67e07294c90bcd9a0ffb038bf8aaad0670775e/src/net_processing.cpp#L2733-L2735) though the last one is what upstream seems to prefer even if I find it unsightly. Was only suggesting that there be an inline comment and wrote one for convenience...